### PR TITLE
Use glib for option processing in crm_ticket, and update the code a bit

### PR DESCRIPTION
**Pull request by clumens**

### CODE REVIEW

File: tools/crm_ticket.c [hunk 1]
```diff
@@ -729,131 +745,29 @@ main(int argc, char **argv)
...
-        }
+    if (!g_option_context_parse_strv(context, &processed_args, &error)) {
+        exit_code = CRM_EX_USAGE;
+        goto done;
     }
 
-    if (optind > argc) {
-        ++argerr;
-    }
+    pcmk__cli_init_logging("attrd_updater", args->verbosity);
```
Comment (kgaillot) [inline tools/crm_ticket.c:764]: copy paste :)

File: tools/crm_ticket.c [hunk 2]
```diff
@@ -729,131 +745,29 @@ main(int argc, char **argv)
...
 
-    if (argerr) {
-        pcmk__cli_help('?', CRM_EX_USAGE);
+    if (args->version) {
+        g_strfreev(processed_args);
+        pcmk__free_arg_context(context);
+        /* FIXME:  When crm_diff is converted to use formatted output, this can go. */
```
Comment (kgaillot) [inline tools/crm_ticket.c:769]: again :)

**Review by kgaillot — COMMENTED** [review]
Nice :)

File: tools/crm_ticket.c (hunk 1 above)
Comment (clumens) [reply tools/crm_ticket.c:764]: Hah, whoops.

File: tools/crm_ticket.c [hunk 3]
```diff
@@ -56,8 +60,208 @@ GHashTable *attr_set;
...
+    options.attr_value = NULL;
+    modified = true;
+
+    return TRUE;
+}
+
+static GOptionEntry required_entries[] = {
+    { "ticket", 't', 0, G_OPTION_ARG_STRING, &options.ticket_id,
+      "Ticket ID",
+      "ID" },
```
[the remark this answers is not in the document]
Comment (clumens) [reply tools/crm_ticket.c:157]: Fixed.

File: tools/crm_ticket.c [hunk 4]
```diff
@@ -56,8 +60,208 @@ GHashTable *attr_set;
...
+attr_value_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **err) {
+    pcmk__str_update(&options.attr_value, optarg);
+
+    if (!options.attr_name || !options.attr_value) {
+        return TRUE;
+    }
+
+    g_hash_table_insert(attr_set, strdup(options.attr_name), strdup(options.attr_value));
+    free(options.attr_name);
+    options.attr_name = NULL;
```
[the remark this answers is not in the document]
Comment (clumens) [reply tools/crm_ticket.c:75]: This comes up in another place as well, so I might as well change it just to simplify the code.

**Pull request merged**